src/models/NavItemPage.php

```diff
@@ -205,24 +206,23 @@ private function renderPlaceholderRecursive($navItemPageId, $placeholderVar, $pr
             $prev = $key-1;
             $next = $key+1;
             $cacheKey = ['blockcache', (int) $placeholder['id']];
-            
+
             $blockResponse = $this->getHasCache($cacheKey);
-            
-            if ($blockResponse === false) {
-                
-                /** @var $blockObject \luya\cms\base\InternalBaseBlock */
-                $blockObject = Block::objectId($placeholder['block_id'], $placeholder['id'], 'frontend', $this->getNavItem());
-                
-                // see if its a valid block object
-                if ($blockObject) {
+    
+            /** @var $blockObject \luya\cms\base\InternalBaseBlock */
+            $blockObject = Block::objectId($placeholder['block_id'], $placeholder['id'], 'frontend', $this->getNavItem());
+    
+            if ($blockObject) {
+                if ($blockResponse === false) {
```

Comment:
@boehsermoe $blockResponse (from cache) should checked before assign `blockObject` otherwise we wont' have the cache effect of the sql statements.

Reply:
I need the `$blockObject` on line 278. `onRegister` must called when block is loaded from cache. So I added a static `onRegisterFromCache` function calling it `PhpBlock::onRegisterFromCache()` so it is not defined in the BlockInterface and can not inherit. 
Maybe cache the block class name of the of the placeholder?